scripts/base/init-bare.zeek

```diff
@@ -5771,6 +5771,129 @@ export {
 	const flowbuffer_contract_threshold = 2 * 1024 * 1024 &redef;
 }
 
+module Telemetry;
+export {
+
+	type MetricType: enum {
+		DOUBLE_COUNTER,
+		INT_COUNTER,
+		DOUBLE_GAUGE,
+		INT_GAUGE,
+		DOUBLE_HISTOGRAM,
+		INT_HISTOGRAM,
+	};
+
+	## Type that captures options used to create metrics.
+	type MetricOpts: record {
+		## The prefix (namespace) of the metric. Will be unset when
+		## returned from :zeek:see:`Telemetry::collect_metrics` or
+		## :zeek:see:`Telemetry::collect_histogram_metrics`.
+		prefix: string &optional;
+
+		## The human-readable name of the metric. This is set to the
+		## full prefixed name including the unit when returned from
+		## :zeek:see:`Telemetry::collect_metrics` or
+		## :zeek:see:`Telemetry::collect_histogram_metrics`.
+		name: string;
+
+		## The unit of the metric. Set to a blank string if this is a
+		## unit-less metric. Will be unset when returned from
+		## :zeek:see:`Telemetry::collect_metrics` or
+		## :zeek:see:`Telemetry::collect_histogram_metrics`.
+		unit: string &optional;
+
+		## Documentation for this metric.
+		help_text: string;
+
+		## The label names (also called dimensions) of the metric. When
+		## instantiating or working with concrete metrics, corresponding
+		## label values have to be provided.
+		labels: vector of string &default=vector();
+
+		## Whether the metric represents something that is accumulating.
+		## Defaults to ``T`` for counters and ``F`` for gauges and
```

Comment:
Fixed up the comments a little bit. I've debated whether these are necessary too. The double fields will have all of the values in them, so why should we return both double and count values if the instrument holds int values. That opens another question though, as to why we bother to have double and int instruments at all and why we don't just have doubles.

Reply:
>  It seems safe to me to deprecate this field (or just remove it outright, given the next point...).

Given how much of the API has changed already, I think we can just remove it. I should add a check to not allow a name or a unit to be `total` though.

> it looks safe to me to reduce this to just having all instruments operating on doubles. If we do that, it brings up the question of how much we let it affect the C++ design around BaseType etc. I could see keeping that as-is, in case we make other customizations in the future?

It seems reasonable to me to just reduce it to doubles. That's all Prometheus operates on in the first place. @neverlord Do you remember why the integer versions were implemented originally? As for `BaseType`, it only exists to make the split of int and double easier so I would vote for merging everything together.